spark2/src/main/java/org/apache/iceberg/spark/source/Reader.java

```diff
@@ -143,8 +143,9 @@
       } catch (IOException ioe) {
         LOG.warn("Failed to get Hadoop Filesystem", ioe);
       }
+      Boolean localityFallback = LOCALITY_WHITELIST_FS.contains(scheme);
       this.localityPreferred = options.get("locality").map(Boolean::parseBoolean)
-          .orElse(LOCALITY_WHITELIST_FS.contains(scheme));
+          .orElse(localityFallback);
```

Comment:
Because schema is non-final we can't just switch this to a lambda

Reply:
I think actually this would actually be fine, but to make the rules happy we would have to reassign Scheme anyway so I figured we may as well just put the result in a local var

Reply:
We could refactor this into a method to make scheme effectively final here.